### DISCUSSION TITLE
Mypy typing issue with text().columns() call

### DISCUSSION THREAD

**Discussion by pamelafox** [sqlalchemy/sqlalchemy#11600] [Usage Questions]
I have this code:

```
sql = text(hybrid_query).columns(id=Integer, score=Float)
```

I am getting this mypy error:

```
error: Argument "rank" to "columns" of "TextClause" has incompatible type "type[Integer]"; expected "TypeEngine[Any]"  [arg-type]
```

My code looks similar to what I see in the SQLAlchemy docs/tests, although I also see there's an alternate synax with column(). What's the right way to write this code, typing-wise?

**Comment by zzzeek** [comment]
I think this might be a shortcoming in our typing that you can work around for now by instantiating those SQL types, like this:

`columns(id=Integer(), score=Float())`

we have a type alias that's supposed to be used for SQL type arguments that accommodates for these two forms.  it's likely not used correctly in this method right now (we should fix it).

**Reply by pamelafox** [reply]
Okay, we've moved to that syntax instead for now, thanks!
